MantidQt/MantidWidgets/src/DataProcessorUI/QDataProcessorTwoLevelTreeModel.cpp

```diff
@@ -467,14 +468,48 @@ void QDataProcessorTwoLevelTreeModel::setupModelData(
 }
 
 /** Return the underlying data structure, i.e. the table workspace this model is
- * representing
- *
- * @return :: the underlying table workspace
- */
+* representing
+* @return :: the underlying table workspace
+*/
 ITableWorkspace_sptr
 QDataProcessorTwoLevelTreeModel::getTableWorkspace() const {
   return m_tWS;
 }
 
+/** Add a new data item to be highlighted
+* @param position : The position of the item to be highlighted
+* @param parent : The parent of this item
+* @return : Boolean indicating whether the row was successfully highlighted
+*/
+bool QDataProcessorTwoLevelTreeModel::addHighlighted(
+    int position, const QModelIndex &parent) {
+
+  if (!parent.isValid()) {
```

Comment:
Yes pretty much. If the index is valid but the parent isn't, it is a child item. If the index is valid and the parent is valid, it is a parent item.